packages/core/helper-plugin/src/features/AutoReloadOverlayBlocker.tsx

```diff
@@ -19,48 +18,43 @@ import { pxToRem } from '../utils/pxToRem';
  * because it's easier to diagnose and we're not using a million refs because we don't
  * understand what's going on.
  */
-
-/**
- * @preserve
- * @typedef {Object} AutoReloadOverlayBlockerConfig
- * @property {string | undefined} title
- * @property {string | undefined} description
- * @property {'reload' | 'time' | undefined} icon
- */
+export interface AutoReloadOverlayBlockerConfig {
+  title?: string;
+  description?: string;
+  icon?: 'reload' | 'time';
+}
 
 /* -------------------------------------------------------------------------------------------------
  * Context
  * -----------------------------------------------------------------------------------------------*/
 
-/**
- * @preserve
- * @typedef {Object} AutoReloadOverlayBlockerContextValue
- * @property {(config: AutoReloadOverlayBlockerConfig) => void} lockAppWithAutoreload
- * @property {() => void} unlockAppWithAutoreload
- */
+type EmptyObject = Record<string, never>;
 
-/**
- * @preserve
- * @type {React.Context<AutoReloadOverlayBlockerContextValue>}
- */
+export interface AutoReloadOverlayBlockerContextValue {
+  lockAppWithAutoreload: (config?: AutoReloadOverlayBlockerConfig) => void;
+  unlockAppWithAutoreload: () => void;
+}
 
-const AutoReloadOverlayBlockerContext = React.createContext();
+const AutoReloadOverlayBlockerContext = React.createContext<
+  AutoReloadOverlayBlockerContextValue | EmptyObject
+>({});
```

Comment:
would it make more sense to make the functions optional? because when we use the code, we'll still have to check if they exist therefore there's no real difference imo